src/standalone/Soil/boundary_conditions.jl

```diff
@@ -13,6 +16,129 @@ An abstract type for soil-specific types of boundary conditions, like free drain
 """
 abstract type AbstractSoilBC <: ClimaLSM.AbstractBC end
 
+
+"""
+    boundary_vars(::NamedTuple, ::ClimaLSM.TopBoundary)
+
+The list of symbols for additional variables to add to the soil
+model auxiliary state, which defaults to adding storage for the
+top and bottom boundary flux fields,
+but which can be extended depending on the type of boundary condition used.
+
+Use this function in the exact same way you would use `auxiliary_vars`.
+"""
+boundary_vars(::NamedTuple, ::ClimaLSM.TopBoundary) = (:top_bc,)
+
+"""
+    boundary_vars(::NamedTuple, ::ClimaLSM.BottomBoundary)
+
+The list of symbols for additional variables to add to the soil
+model auxiliary state, which defaults to adding storage for the
+top and bottom boundary flux fields,
+but which can be extended depending on the type of boundary condition used.
+
+Use this function in the exact same way you would use `auxiliary_vars`.
+"""
+boundary_vars(::NamedTuple, ::ClimaLSM.BottomBoundary) = (:bottom_bc,)
+
+"""
+    bc_var_domain_names(::AbstractSoilBC, ::ClimaLSM.AbstractBoundary)
+
+The list of domain names for additional variables to add to the soil
+model auxiliary state,  which defaults to adding storage for the
+top and bottom boundary flux fields,
+but which can be extended depending on the type of boundary condition used.
+
+Use this function in the exact same way you would use `auxiliary_var_domain_names`.
+"""
+boundary_var_domain_names(::NamedTuple, ::ClimaLSM.AbstractBoundary) =
+    (:surface,)
+
+"""
+    boundary_var_types(::Soil.EnergyHydrology{FT}, ::NamedTuple, ::ClimaLSM.AbstractBoundary) where {FT}
+
+The list of variable types for additional variables to add to the soil
+model auxiliary state, which defaults to adding storage for the
+top and bottom boundary flux fields,
+but which can be extended depending on the type of boundary condition used.
+
+Use this function in the exact same way you would use `auxiliary_types`.
+"""
+function boundary_var_types(
+    ::Soil.EnergyHydrology{FT},
+    ::NamedTuple,
+    ::ClimaLSM.AbstractBoundary,
+) where {FT}
+    (NamedTuple{(:water, :heat), Tuple{FT, FT}},)
+end
+
+"""
+    boundary_var_types(::Soil.RichardsModel{FT}, ::NamedTuple, ::ClimaLSM.AbstractBoundary) where {FT}
+
+The list of variable types for additional variables to add to the soil
+model auxiliary state, which defaults to adding storage for the
+top and bottom boundary flux fields,
+but which can be extended depending on the type of boundary condition used.
+
+Use this function in the exact same way you would use `auxiliary_types`.
+"""
+boundary_var_types(
+    ::Soil.RichardsModel{FT},
+    ::NamedTuple,
+    ::ClimaLSM.AbstractBoundary,
+) where {FT} = (FT,)
```

Comment:
potentially? I could go either way on this. Right now these are the two use cases. I think if we made this NamedTuple typed, it would just be for consistency. We would use it like:
```
    p.soil.top_bc.water .= boundary_flux(
            model.boundary_conditions.top.water,
            TopBoundary(),
            model,
            Δz_top,
            Y,
            p,
            t,
        )
```

instead of 
```
    p.soil.top_bc .= boundary_flux(
            model.boundary_conditions.top.water,
            TopBoundary(),
            model,
            Δz_top,
            Y,
            p,
            t,
        )
```

but I think there would be little else that changed. what is your preference?

Reply:
for now, I added to the doc string explaining why this is a scalar while the EnergyHydrology method returns a NamedTuple.

Reply:
I think youre right! we also pass in the BC as NamedTuples, so we should do that here. ill make this change.